packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -762,6 +773,285 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
+        namespace,
+        force,
+      });
+    // bulk up the bulkDeleteParams
+    expectedBulkDeleteMultiNamespaceDocsResults.map((expectedResult) => {
+      if (isRight(expectedResult)) {
+        bulkDeleteParams.push({
+          delete: {
+            _id: this._serializer.generateRawId(
+              namespace,
+              expectedResult.value.type,
+              expectedResult.value.id
+            ),
+            _index: this.getIndexForType(expectedResult.value.type),
+            ...getExpectedVersionProperties(undefined),
+          },
+        });
+      }
+    });
+
+    const bulkDeleteResponse = bulkDeleteParams.length
+      ? await this.client.bulk({
+          refresh,
+          body: bulkDeleteParams,
+          require_alias: true,
+        })
+      : undefined;
+
+    // extracted to ensure consistency in the error results returned
+    let errorResult: BulkDeleteItemErrorResult;
+    const savedObjects = await Promise.all(
+      expectedBulkDeleteMultiNamespaceDocsResults.map(async (expectedResult) => {
+        if (isLeft(expectedResult)) {
+          return { ...expectedResult.value, success: false };
+        }
+        const {
+          type,
+          id,
+          namespaces,
+          esRequestIndex: esBulkDeleteRequestIndex,
+        } = expectedResult.value;
+        // we assume this wouldn't happen but is needed to ensure type consistency
+        if (bulkDeleteResponse === undefined) {
+          throw new Error(
+            `Unexpected error in bulkDelete saved objects: bulkDeleteResponse is undefined`
+          );
+        }
+        const rawResponse = Object.values(
+          bulkDeleteResponse.items[esBulkDeleteRequestIndex]
+        )[0] as NewBulkItemResponse;
+
+        const error = getBulkOperationError(type, id, rawResponse);
+        if (error) {
+          errorResult = { success: false, type, id, error };
+          return errorResult;
+        }
+        if (rawResponse.result === 'not_found') {
+          errorResult = {
+            success: false,
+            type,
+            id,
+            error: errorContent(SavedObjectsErrorHelpers.createGenericNotFoundError(type, id)),
+          };
+          return errorResult;
+        }
+
+        if (rawResponse.result === 'deleted') {
+          // `namespaces` should only exist in the expectedResult.value if the type is multi-namespace.
+          if (namespaces) {
+            // in the bulk operation, one cannot specify a namespace from which to delete an object other than the namespace that the operation is performed in.
+            // If a multinamespace object exists in more than the current space (from which the call is made), force deleting the object will delete it from all namespaces it exists in.
+            // In that case, all legacy url aliases are deleted as well. If force isn't applied, the operation fails and the object isn't deleted.
+            await deleteLegacyUrlAliases({
```

Comment:
`deleteLegacyUrlAliases` is asynchronous and only accepts one object at a time, hence having to use `Promise.all` here. We could, potentially, follow a similar strategy to how the `bulkDeleteParams` are created and postpone the call to delete legacy URL aliases before returning the final `statuses` response.

We're not too concerned with the final result from all the calls to `deleteLegacyUrlAliases` so would we still need to wait for the promises to resolve or use `Promise.allSettled` instead?